src/commands.ts

```diff
@@ -21,27 +24,46 @@ export function registerCommands(
     }
   });
 
-  vscode.commands.registerCommand("semgrep.scan", async (file?: vscode.Uri) => {
-    let uri = null;
-    if (!file) {
-      // Get active file if no path provided
-      const activeEditor = vscode.window.activeTextEditor;
-      if (activeEditor) {
-        uri = activeEditor.document.uri;
-      }
-    } else {
-      uri = file;
-    }
-    if (uri) {
-      await client.sendNotification(scan, { uri: uri.toString() });
-    } else {
-      vscode.window.showWarningMessage(
-        "No file provided to scan, please open a file or provide a file name"
-      );
+  vscode.commands.registerCommand("semgrep.scanWorkspace", async () => {
+    await client.sendNotification(scanWorkspace, { full: false });
+  });
+
+  vscode.commands.registerCommand("semgrep.scanWorkspaceFull", async () => {
+    await client.sendNotification(scanWorkspace, { full: true });
+  });
+
+  vscode.commands.registerCommand("semgrep.logout", async () => {
+    await client.sendNotification(logout);
+    env.loggedIn = false;
+  });
+
+  vscode.commands.registerCommand("semgrep.refreshRules", async () => {
+    await client.sendNotification(refreshRules);
+  });
+
+  vscode.commands.registerCommand("semgrep.loginStatus", async () => {
+    const result: LoginStatusParams | null = await client.sendRequest(
+      loginStatus
+    );
+    if (result) {
+      env.logger.log("Status Result " + result.loggedIn);
+      env.loggedIn = result.loggedIn;
     }
   });
 
-  vscode.commands.registerCommand("semgrep.scanWorkspace", async () => {
-    await client.sendNotification(scanWorkspace);
+  vscode.commands.registerCommand("semgrep.loginNudge", async () => {
+    env.logger.log("logged in: " + env.loggedIn);
+    if (!env.loggedIn && env.showNudges) {
+      const resp = await vscode.window.showInformationMessage(
+        "Login to enable additional proprietary Semgrep Registry rules and running custom policies from Semgrep Code",
+        "Login",
```

Comment:
```suggestion
        "Sign in",
```

Just using the same words as on https://semgrep.dev/login